### PR TITLE
fix exception if set duty_cycle before period

### PR DESCRIPTION
**Pull request by suriyan**

### CODE REVIEW

File: rpi_hardware_pwm/__init__.py [hunk 1]
```diff
@@ -109,7 +109,8 @@ def change_frequency(self, hz: float):
 
         # we first have to change duty cycle, since https://stackoverflow.com/a/23050835/1895939
         original_duty_cycle = self._duty_cycle
-        self.change_duty_cycle(0)
+        if self._duty_cycle:
+            self.change_duty_cycle(0)
```
Comment (CamDavidsonPilon) [inline rpi_hardware_pwm/__init__.py:113]: 👋 

Hm, how could duty_cycle be set before frequency? Can you post a reproducible example where this is failing? 

(btw your change is how the code looked before (a few days ago), and I thought I got around using this `if` statement, but I may have missed something)

Comment (suriyan) [reply rpi_hardware_pwm/__init__.py:113]: The following is the example script to reproduce the exception:-

```
Python 3.7.3 (default, Dec 20 2019, 18:57:59)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from rpi_hardware_pwm import HardwarePWM
>>> pwm = HardwarePWM(0, hz=60)
OSError: [Errno 22] Invalid argument

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/pi/devel/rpi_hardware_pwm/rpi_hardware_pwm/__init__.py", line 59, in __init__
    self.change_frequency(hz)
  File "/home/pi/devel/rpi_hardware_pwm/rpi_hardware_pwm/__init__.py", line 112, in change_frequency
    self.change_duty_cycle(0)
  File "/home/pi/devel/rpi_hardware_pwm/rpi_hardware_pwm/__init__.py", line 102, in change_duty_cycle
    self.echo(dc, os.path.join(self.pwm_dir, "duty_cycle"))
  File "/home/pi/devel/rpi_hardware_pwm/rpi_hardware_pwm/__init__.py", line 76, in echo
    f.write(f"{message}\n")
OSError: [Errno 22] Invalid argument
>>>
```

We will get the exception only for the first time use after export it:-

```
# echo 0 > /sys/class/pwm/pwmchip0/export
# echo 0 > /sys/class/pwm/pwmchip0/pwm0/duty_cycle
-bash: echo: write error: Invalid argument
```

But it's okay if we set `period` before `duty_cycle`:-

```
# echo 1 > /sys/class/pwm/pwmchip0/export
# echo 100000 > /sys/class/pwm/pwmchip0/pwm1/period
# echo 0 > /sys/class/pwm/pwmchip0/pwm1/duty_cycle
```

Comment (CamDavidsonPilon) [reply rpi_hardware_pwm/__init__.py:113]: You're right! I had some leftover state on my machine that was messing with my tests. I'll address this asap after a few more tests

Comment (CamDavidsonPilon) [reply rpi_hardware_pwm/__init__.py:113]: Your change makes sense. I'll merge and upload a new version to PyPi

**Comment by CamDavidsonPilon** [comment]
New version uploaded to PyPI

**Pull request merged**